### PR TITLE
zeroconf: Implement new approach for ip detection

### PR DESCRIPTION
**Pull request by cmroche**
GitHub is for some reason showing the old PR as merge, this is an error, and it won't let me restore so I need to create a new PR: https://github.com/Arksine/moonraker/pull/284

However this implements the latest and greatest approach to getting and updating IP address for the Zeroconf service.

### CODE REVIEW

**Comment by cmroche** [comment]
@Arksine  Sorry to fracture the conversation here, GitHub refuses to let me use the old PR any longer. But I think this should do it.

File: moonraker/components/zeroconf.py [hunk 1]
```diff
@@ -32,29 +40,65 @@ async def unregister_services(self, infos: List[AsyncServiceInfo]) -> None:
         await asyncio.gather(*background_tasks)
         await self.aiozc.async_close()
 
-class ZeroconfRegistrar():
+    async def update_services(self, infos: List[AsyncServiceInfo]) -> None:
+        assert self.aiozc is not None
+        tasks = [self.aiozc.async_update_service(info) for info in infos]
+        background_tasks = await asyncio.gather(*tasks)
+        await asyncio.gather(*background_tasks)
+        await self.aiozc.async_close()
```
Comment (Arksine) [inline moonraker/components/zeroconf.py:48]: Should `async_close()` be called here?  I would think that this would shut down the zeroconf.

**Comment by Arksine** [comment]
No problem, Github probably just didn't like the merge conflict.  For the most part everything looks good, just need to check on the `async_close()` call.

File: moonraker/components/zeroconf.py (hunk 1 above)
Comment (cmroche) [reply moonraker/components/zeroconf.py:48]: Nope, missing commit.. one moment.

**Pull request merged**